addons/sale_project_account/models/account_move.py

```diff
@@ -0,0 +1,11 @@
+# -*- coding: utf-8 -*-
+
+from odoo import models
+
+
+class AccountMoveLine(models.Model):
+    _inherit = 'account.move.line'
+
+    def _compute_analytic_account_id(self):
+        filtered_set = self.filtered(lambda aml: not (aml.analytic_account_id and any(aml.sale_line_ids.project_id)))
+        super(AccountMoveLine, filtered_set)._compute_analytic_account_id()
```

Comment:
For readability I would make the filter condition positive 
```
        project_amls = self.filtered(lambda aml: aml.analytic_account_id and aml.sale_line_ids.project_id)
        super(AccountMoveLine, self - project_amls)._compute_analytic_account_id()
```
Also, a comment on why we are doing this would be nice